core/src/main/java/net/opentsdb/data/types/numeric/UglyByteNumericSerdes.java

```diff
@@ -60,43 +60,50 @@ public void serialize(final TimeSeriesQuery query,
     try {
       final TimeSeriesId id = data.id();
       
-      stream.write(Bytes.fromInt(id.alias() == null ? 0 : id.alias().length));
-      if (id.alias() != null && id.alias().length > 0) {
-        stream.write(id.alias());
+      byte[] buf = id.alias() == null ? null : id.alias().getBytes(Const.UTF8_CHARSET);
```

Comment:
Pick a better name for the class?

Also, can we replace this encoding with gorilla encoding?

Reply:
heh, yeah we definitely can and should replace this with gorilla encoding. It's just a sample class.